src/openassetio-core/hostApi/ManagerFactory.cpp

```diff
@@ -66,6 +74,69 @@ ManagerPtr ManagerFactory::createManagerForInterface(
       managerImplementationFactory->instantiate(identifier),
       managerApi::HostSession::make(managerApi::Host::make(hostInterface), logger));
 }
+
+ManagerPtr ManagerFactory::defaultManagerForInterface(
+    const HostInterfacePtr& hostInterface,
+    const ManagerImplementationFactoryInterfacePtr& managerImplementationFactory,
+    const log::LoggerInterfacePtr& logger) {
+  const char* configPath = std::getenv(kDefaultManagerConfigEnvVarName.c_str());
+
+  if (!configPath) {
+    Str msg = kDefaultManagerConfigEnvVarName + " not set, unable to instantiate default manager.";
+    // We leave this as a debug message, as it is expected may hosts
+    // will call this by default, and handle a null return manager, vs
+    // it being a warning/error.
+    logger->log(log::LoggerInterface::Severity::kDebug, msg);
+    return nullptr;
+  }
+
+  {
+    Str msg = "Loading default manager config from '";
+    msg += configPath;
+    msg += "' [" + kDefaultManagerConfigEnvVarName + "]";
+    logger->log(log::LoggerInterface::Severity::kDebug, msg);
+  }
+
+  if (!std::filesystem::exists(configPath)) {
+    Str msg = "Could not load default manager config from '";
+    msg += configPath;
+    msg += "', file does not exist.";
+    throw std::runtime_error(msg);
+  }
+
+  auto config = toml::parse_file(configPath);
+
+  std::string_view identifier = config["manager"]["identifier"].value_or("");
+
+  InfoDictionary settings;
+  if (toml::table* settingsTable = config["manager"]["settings"].as_table()) {
+    for (auto&& [key, val] : *settingsTable) {
```

Comment:
[consider][minor] Make use of [`for_each`](https://marzer.github.io/tomlplusplus/classtoml_1_1table.html#a8a73c50b751f4e20dedfeb14a80f9381) visitor. It looks like you could avoid the `if` block altogether... OK not altogether, since there are types we don't support.

Otherwise, the double `&&` looks a bit odd, though apparently it doesn't cause any problems.  I would probably change this to `const auto&`

Reply:
Improvements made [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/608/commits/dc5a5eaa1107dd9ea23396a42548de39fb6143ea) based on off-line chat. The short is we can't use the `for_each` visitor as filtering the types to avoid "no matching initializer" ends up being more verbose than this approach.